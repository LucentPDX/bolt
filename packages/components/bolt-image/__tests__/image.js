const { render } = require('@bolt/twig-renderer');
const { readYamlFileSync } = require('@bolt/build-tools/utils/yaml');
const { join } = require('path');
const schema = readYamlFileSync(join(__dirname, '../image.schema.yml'));

describe('<bolt-image> Component', async () => {
  test('<bolt-image> with ratio object compiles', async () => {
    const results = await render('@bolt-components-image/image.twig', {
      src: '/fixtures/1200x660.jpg',
      alt: 'A Rock Climber',
      lazyload: true,
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('<bolt-image> with ratio object and lazyloading disabled compiles', async () => {
    const results = await render('@bolt-components-image/image.twig', {
      src: '/fixtures/1200x660.jpg',
      alt: 'A Rock Climber',
      lazyload: false,
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
