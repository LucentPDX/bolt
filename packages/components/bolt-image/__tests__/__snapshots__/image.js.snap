// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-image> Component <bolt-image> with ratio object and lazyloading disabled compiles 1`] = `

<bolt-image style="background-color: hsl(233, 33%, 97%);">
  <bolt-ratio aspect-ratio-height="660"
              aspect-ratio-width="1200"
              style="--aspect-ratio-height: 660; --aspect-ratio-width: 1200;"
  >
    <img class="c-bolt-image__image-placeholder"
         alt="A Rock Climber"
         sizes="auto"
         src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
    >
    <img class="c-bolt-image__image"
         alt="A Rock Climber"
         srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w"
         sizes="auto"
         src="/fixtures/1200x660.jpg"
    >
  </bolt-ratio>
</bolt-image>

`;

exports[`<bolt-image> Component <bolt-image> with ratio object compiles 1`] = `

<bolt-image style="background-color: hsl(233, 33%, 97%);">
  <bolt-ratio aspect-ratio-height="660"
              aspect-ratio-width="1200"
              style="--aspect-ratio-height: 660; --aspect-ratio-width: 1200;"
  >
    <img class="c-bolt-image__image-placeholder"
         alt="A Rock Climber"
         src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
    >
    <img class="c-bolt-image__image c-bolt-image__lazyload c-bolt-image__lazyload--fade c-bolt-image__lazyload--blur js-lazyload"
         alt="A Rock Climber"
         data-srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w"
         data-sizes="auto"
         src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
    >
    <noscript>
      <img class="c-bolt-image__image"
           alt="A Rock Climber"
           src="/fixtures/1200x660.jpg"
           srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w"
      >
    </noscript>
  </bolt-ratio>
</bolt-image>

`;
